torch_geometric/loader/dataloader.py

```diff
@@ -39,6 +40,29 @@ def collate(self, batch):  # Deprecated...
         return self(batch)
 
 
+# PyG 'Data' objects are subclasses of MutableMapping, which is an
+# instance of collections.abc.Mapping. Currently, PyTorch pin_memory
+# for DataLoaders treats the returned batches as Mapping objects and
+# calls `pin_memory` on each element in `Data.__dict__`, which is not
+# desired behavior if 'Data' has a `pin_memory` function. We patch
+# this behavior here by monkeypatching `pin_memory`, but can hopefully patch
+# this in PyTorch in the future:
+__torch_pin_memory = torch.utils.data._utils.pin_memory.pin_memory
+
+
+def pin_memory(data, device=None):
+    if hasattr(data, "pin_memory"):
+        return data.pin_memory()
+    else:
+        if len(signature(__torch_pin_memory).parameters) > 1:
```

Comment:
Can we pre-compute this? AFAIK, `signature` is a pretty slow call.

Reply:
nice catch